dra-evolution/pkg/api/capacity_types.go

```diff
@@ -64,12 +64,33 @@ type ResourcePoolSpec struct {
 	// Must not have more than 128 entries.
 	Devices []Device `json:"devices,omitempty"`
 
+	// SharedCapacity are pooled resources that are shared by all
+	// devices in the pool. This is typically used when representing a
+	// partitionable device, and need not be populated otherwise.
+	//
+	// Must not have more than 32 entries.
+	//
+	// +optional
+	// +listType=atomic
+	SharedCapacity []SharedCapacity `json:"sharedCapacity,omitempty"`
+
 	// FUTURE EXTENSION: some other kind of list, should we ever need it.
 	// Old clients seeing an empty Devices field can safely ignore the (to
 	// them) empty pool.
 }
 
+type ResourceCapacity struct {
+	// Name is the resource name/type.
+	// +required
+	Name string `json:"name"`
```

Comment:
I assume all of the `Name` fields should be a DNS label?

Reply:
Is there mismatch between the type names? Above it's `SharedCapacity`, here it's `ResourceCapacity`.

Reply:
I am flexible on the various names. `SharedCapacity` is a named list of `ResourceCapacity` objects.

Reply:
> I assume all of the Name fields should be a DNS label?

Can we make the validation identical to the identifier part of an attribute name? That's merely for consistency, not because the two are related. As explained elsewhere, these shared capacity  names are not exposed to users.